src/_guides/language/concurrency/index.md

```diff
@@ -312,107 +311,144 @@ to implement isolates.
+its result is always returned as a `Future`.
+The original isolate doesn't care whether the computation
+running in the other isolate is synchronous or asynchronous,
+because it's being run concurrently either way.
 
-<?code-excerpt "lib/simple_worker_isolate.dart (spawned)"?>
+{% comment %}
+TODO:
+Should create a diagram for the current example.
+Previous example's diagram and text for reference:
+
+  The following figure illustrates the communication between
+  the main isolate and the worker isolate:
+  
+  ![A figure showing the previous snippets of code running in the main isolate and in the worker isolate](/guides/language/concurrency/images/isolate-api.png)
+{% endcomment %}
+
+#### Sending closures with isolates
+
+This section shows another simple worker isolate created with `run()`,
+this time written as a function literal, or closure,
+directly in the main isolate.
+
+<?code-excerpt "lib/simple_isolate_closure.dart (main)"?>
 ```dart
-Future<void> _readAndParseJson(SendPort p) async {
-  final fileData = await File(filename).readAsString();
-  final jsonData = jsonDecode(fileData);
-  Isolate.exit(p, jsonData);
+void main() async {
+  // Read some data.
+  final jsonData = await Isolate.run(() async {
+    final fileData = await File(filename).readAsString();
+    final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+    return jsonData;
+  });
+
+  // Use that data.
+  print('Number of JSON keys: ${jsonData.length}');
 }
 ```
 
-The relevant statement is the last one, which exits the isolate,
-sending `jsonData` to the passed-in `SendPort`.
-Message passing using `SendPort.send` normally involves data copying,
-and thus can be slow.
-However, when you send the data using `Isolate.exit()`,
-then the memory that holds the message in the exiting isolate isn’t copied,
-but instead is transferred to the receiving isolate.
-The sender will nonetheless perform a verification pass to ensure
-the objects are allowed to be transferred.
-
-
-{{site.alert.version-note}}
-  `Isolate.exit()` was added in 2.15.
-  Previous releases support only explicit message passing,
-  using `Isolate.send()` as shown in the next section's example.
-{{site.alert.end}}
-
-The following figure illustrates the communication between
-the main isolate and the worker isolate:
+This example accomplishes the same as the previous:
+a new isolate spawns and executes a computation, sending back the result.
 
-![A figure showing the previous snippets of code running in the main isolate and in the worker isolate](/guides/language/concurrency/images/isolate-api.png)
+However, now the isolate sends a [closure][]. 
+This allows versatility not only for how the isolate can be utilized
+(as closures are less limited than typical named functions),
+but also for the written style of the code.
+Here, `Isolate.run()` executes what looks like local code, concurrently.
+In that sense, you can imagine `run()` almost like a [control flow operator][]
+for “run in parallel”.
```

Comment:
**suggestion:** This could also use tightening.

```suggestion
Now the isolate returns a [closure][]. 
Closures provide more options as to how to use the isolate
and how to write the code.
They have fewer limits than named functions.
In this example, `Isolate.run()` executes what looks like
local code in a concurrent manner.
In that sense, imagine `run()` to work like a [control flow operator][]
for “run in parallel”.
```

Reply:
Is it a hard-must to get rid of every "However"? I do overuse it sometimes, but I feel like in this case it's somewhat warranted... 

Reading the line preceding this paragraph, and then following with "Now" feels abrupt and reads like an incomplete thought. 

> This example accomplishes the same as the previous. A new isolate spawns, computes something, and sends back the result.
> 
> Now the isolate sends a closure

Like, it doesn't sound like we're talking about the difference between the previous example and this one.
I want to keep the emphasis on the _difference_, not just stating what's happening for the sake of stating it. Is there another way to do that "feels" smoother?